conda_forge_tick/auto_tick.py

```diff
@@ -0,0 +1,994 @@
+    load_graph,
+    dump_graph,
+    LazyJson,
+)
+from .xonsh_utils import env
+from typing import (
+    List,
+    Optional,
+    MutableSequence,
+    Sequence,
+    Tuple,
+    Dict,
+    Set,
+    MutableMapping,
+)
+
+logger = logging.getLogger("conda_forge_tick.auto_tick")
+
+# TODO: move this back to the bot file as soon as the source issue is sorted
+# https://travis-ci.org/regro/00-find-feedstocks/jobs/388387895#L1870
+from conda_forge_tick.utils import frozen_to_json_friendly
+from conda_forge_tick.contexts import MigratorsContext
+from conda_forge_tick.migrators import (
+    Migrator,
+    Version,
+    PipMigrator,
+    LicenseMigrator,
+    MigrationYaml,
+    MigratorContext,
+    Rebuild,
+    Replacement,
+    ArchRebuild,
+)
+
+from .migrators_types import *
+
+if typing.TYPE_CHECKING:
+    from .cli import CLIArgs
+
+
+MIGRATORS: MutableSequence[Migrator] = [
+    Version(pr_limit=30, piggy_back_migrations=[PipMigrator(), LicenseMigrator()]),
+    # Noarch(pr_limit=10),
+    # Pinning(pr_limit=1, removals={'perl'}),
+    # Compiler(pr_limit=7),
+]
+
+BOT_RERUN_LABEL = {
+    "name": "bot-rerun",
+    "color": "#191970",
+    "description": "Apply this label if you want the bot to retry issueing a particular pull-request",
+}
+
+
+def run(
+    feedstock_ctx: FeedstockContext,
+    migrator: Migrator,
+    protocol: str = "ssh",
+    pull_request: bool = True,
+    rerender: bool = True,
+    fork: bool = True,
+    **kwargs: typing.Any,
+) -> Tuple["MigrationUidTypedDict", dict]:
+    """For a given feedstock and migration run the migration
+
+    Parameters
+    ----------
+    feedstock_ctx: FeedstockContext
+        The node attributes
+    migrator: Migrator instance
+        The migrator to run on the feedstock
+    protocol : str, optional
+        The git protocol to use, defaults to ``ssh``
+    pull_request : bool, optional
+        If true issue pull request, defaults to true
+    rerender : bool
+        Whether to rerender
+    fork : bool
+        If true create a fork, defaults to true
+    gh : github3.GitHub instance, optional
+        Object for communicating with GitHub, if None build from $USERNAME
+        and $PASSWORD, defaults to None
+    kwargs: dict
+        The key word arguments to pass to the migrator
+
+    Returns
+    -------
+    migrate_return: namedtuple
+        The migration return dict used for tracking finished migrations
+    pr_json: str
+        The PR json object for recreating the PR as needed
+
+    """
+    # get the repo
+    migrator.attrs = feedstock_ctx.attrs  # type: ignore
+    feedstock_dir, repo = get_repo(
+        ctx=migrator.ctx.parent,
+        fctx=feedstock_ctx,
+        branch=migrator.remote_branch(feedstock_ctx),
+        feedstock=feedstock_ctx.feedstock_name,
+        protocol=protocol,
+        pull_request=pull_request,
+        fork=fork,
+    )
+
+    recipe_dir = os.path.join(feedstock_dir, "recipe")
+    # if postscript/activate no noarch
```

Comment:
This seems like it should be part of the noarch migrator. Maybe we should should look up these scripts when we go get he feedstock attrs.